sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -315,6 +315,57 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSparkSession {
     checkAnswer(df.select(isnotnull(col("a"))), Seq(Row(false)))
   }
 
+  test("Test NULLIF function") {
+
+    def testNullIfFunction(alwaysInlineCommonExpr: String): Unit = {
+      withSQLConf(SQLConf.ALWAYS_INLINE_COMMON_EXPR.key -> alwaysInlineCommonExpr) {
+        val spark = SparkSession.builder().getOrCreate()
+
+        val result1 = spark.sql("SELECT NULLIF(1, 1)").collect().head.get(0)
+        assert(result1 == null)
+
+        val result2 = spark.sql("SELECT NULLIF(1, 2)").collect().head.get(0)
+        assert(result2 == 1)
+
+        val result3 = spark.sql("SELECT NULLIF(NULL, 1)").collect().head.get(0)
+        assert(result3 == null)
+
+        val result4 = spark.sql("SELECT NULLIF(1, NULL)").collect().head.get(0)
+        assert(result4 == 1)
+
+        val result5 = spark.sql("SELECT NULLIF(NULL, NULL)").collect().head.get(0)
+        assert(result5 == null)
+
+        val result6 = spark.sql("SELECT NULLIF('abc', 'abc')").collect().head.get(0)
+        assert(result6 == null)
+
+        val result7 = spark.sql("SELECT NULLIF('abc', 'xyz')").collect().head.get(0)
+        assert(result7 == "abc")
```

Comment:
Since all the check above looks similar, let's rewrite it as:
```scala
    Seq(
      "SELECT NULLIF(1, 1)" -> Row(null),
      "SELECT NULLIF(1, 2)" -> Row(1)
    ).foreach { case (sqlText, expectedResult) =>
      checkAnswer(sql(sqlText), expectedResult)
    }
```